src/reconcile/rec_visibility.c

```diff
@@ -69,8 +69,16 @@ __rec_append_orig_value(
 
     /* Review the current update list, checking conditions that mean no work is needed. */
     for (;; upd = upd->next) {
-        /* Done if the update was restored from the history store. */
-        if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_HS))
+        if (upd->txnid != WT_TXN_ABORTED)
+            oldest_upd = upd;
+        /* Leave reference pointing to the last item in the update list. */
+        else if (upd->next == NULL)
+            break;
+        else
+            continue;
+
+        /* Done if the update was restored from the data store or history store. */
+        if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DS | WT_UPDATE_RESTORED_FROM_HS))
             return (0);
 
         /*
```

Comment:
I think there is no need to check the update type to load the on-disk if it is a prepared tombstone because we have already had an update in the update list. Just check for whether the cell is prepared or not?

Reply:
yes, that's correct.

Reply:
Moved the check to the caller.